File: sdk/typescript/examples/basic-chat.ts
/**
 * Basic chat example using dchat TypeScript SDK
 * 
 * Run with: npm run build && node dist/examples/basic-chat.js
 */

import { Client } from '../src';

async function main() {
  console.log('🚀 dchat SDK Basic Chat Example\n');

  // Create a client with the builder pattern
  const alice = await Client.builder()
    .name('Alice')
    .dataDir('/tmp/dchat_alice')
    .listenPort(9001)
    .encryption(true)
    .build();

  const identity = alice.getIdentity();
  console.log(`✅ Client created for: ${identity.username}`);
  console.log(`📍 User ID: ${identity.userId}`);
  console.log(`🔑 Public key: ${identity.publicKey}\n`);

  // Connect to the network
  console.log('🌐 Connecting to dchat network...');
  await alice.connect();
  console.log('✅ Connected!\n');

  // Send a message
  console.log('📤 Sending message...');
  await alice.sendMessage('Hello, decentralized world!');
  console.log('✅ Message sent!\n');

  // Receive messages
  console.log('📥 Fetching messages...');
  const messages = await alice.receiveMessages();
  console.log(`✅ Received ${messages.length} message(s)\n`);

  for (let i = 0; i < messages.length; i++) {
    const msg = messages[i];
    if (msg.content.type === 'Text') {
      console.log(`Message #${i + 1}: ${msg.content.text}`);
    }
  }

  // Disconnect
  console.log('\n🔌 Disconnecting...');
  await alice.disconnect();
  console.log('✅ Disconnected cleanly\n');

  console.log('🎉 Example completed successfully!');
}

main().catch((error) => {
  console.error('❌ Error:', error.message);
  process.exit(1);
});

File: crates/dchat-sdk-rust/examples/basic_chat.rs
//! Basic chat example using dchat Rust SDK
//!
//! Run with: cargo run --package dchat-sdk-rust --example basic_chat

use dchat_sdk_rust::{Client, Result};

#[tokio::main]
async fn main() -> Result<()> {
    println!("🚀 dchat SDK Basic Chat Example\n");

    // Create a client with the builder pattern
    let alice = Client::builder()
        .name("Alice")
        .data_dir("/tmp/dchat_alice")
        .listen_port(9001)
        .encryption(true)
        .build()
        .await?;

    println!("✅ Client created for: {}", alice.identity().username);
    println!("📍 User ID: {}", alice.identity().user_id);
    println!("🔑 Public key fingerprint: {}\n", alice.identity().fingerprint());

    // Connect to the network
    println!("🌐 Connecting to dchat network...");
    alice.connect().await?;
    println!("✅ Connected!\n");

    // Send a message
    println!("📤 Sending message...");
    alice.send_message("Hello, decentralized world!").await?;
    println!("✅ Message sent!\n");

    // Receive messages
    println!("📥 Fetching messages...");
    let messages = alice.receive_messages().await?;
    println!("✅ Received {} message(s)\n", messages.len());

    for (i, msg) in messages.iter().enumerate() {
        println!("Message #{}: {:?}", i + 1, msg.content);
    }

    // Disconnect
    println!("\n🔌 Disconnecting...");
    alice.disconnect().await?;
    println!("✅ Disconnected cleanly\n");

    println!("🎉 Example completed successfully!");

    Ok(())
}
